src/public/modules/forms/services/submissions.client.factory.js

```diff
@@ -232,6 +236,14 @@ function SubmissionsFactory(
         return FormSgSdk.crypto.decryptFile(secretKey, data.encryptedFile)
       })
     },
+    /**
+     * Cancels an existing on-going download
+     */
+    cancelDownloadEncryptedResponses: function () {
+      downloadAbortController.abort()
+      workerPool.forEach((worker) => worker.terminate())
+      workerPool.length = 0 // resets the array
```

Comment:
Amazing, I would have gone with the typical `workerPool.length = []` but this behavior is actually guaranteed by JS in the [ES5 spec](https://es5.github.io/x15.4.html#x15.4).

> The value of the length property is numerically greater than the name of every property whose name is an array index; whenever a property of an Array object is created or changed, other properties are adjusted as necessary to maintain this invariant. Specifically, whenever a property is added whose name is an array index, the length property is changed, if necessary, to be one more than the numeric value of that array index; and whenever the length property is changed, every property whose name is an array index whose value is not smaller than the new length is automatically deleted.

Reply:
Haha `workerPool = []` makes a lot more sense, not sure what I was thinking lol